content/blog/gargle-1-2-0/index.Rmd

```diff
@@ -0,0 +1,243 @@
+
+Various incremental improvements have been made around gargle's token cache:
+
+* The default cache location has moved, to better align with general conventions
+  around where to cache user data. After upgrading gargle, don't be surprised
+  if you see some messages about cache relocation.
+* Token storage relies on serialized R objects and therefore is potentially
+  sensitive to differences in R version and operating system between when a
+  token is created and when it is (re)used. gargle is now able to recognize and
+  cope with the most predictable and harmless effects of moving a token across
+  time or space.
+* gargle is quite conservative about using cached tokens and likes to have
+  explicit permission from the user. However, this is in tension with the desire
+  to make things "just work". Balancing these two goals is an ongoing effort:
+  - In a non-interactive context, gargle will use a cached token, if it can
+    find (at least) one, even if the user has not given explicit instructions.
+    However, we complain noisily and urge the user to make their intent
+    unambiguous in the code. This comes up, for example, when rendering R
+    Markdown documents.
+  - You can now identify the token you want to use with a domain-only glob-like
+    pattern. This makes it possible to successfully execute code like this,
+    non-interactively, on the machines of both `alice@example.com` and
+    `bob@example.com`.
+    
+    ```{r, eval = FALSE}
+    library(googledrive)
+    
+    # googledrive uses gargle for auth
+    drive_auth(email = "*@example.com")
+    ```
+
+## Rolling the built-in OAuth client
+
+User tokens created with gargle are the result of so-called "three-legged OAuth", where the three legs are the user, Google, and an OAuth client (id and secret).
+Each user or application potentially needs to set up their own OAuth client, in order to execute this flow.
+
+A small set of packages maintained by the tidyverse team make use of a shared, built-in OAuth client, which helps new users get started quickly, without the need to create their own client in order to do "Hello, World!".
+Only these specific packages are allowed to use this client and these packages do so according to a specific [privacy policy](https://www.tidyverse.org/google_privacy_policy/).
+The "Tidyverse API Packages" are [bigrquery](https://bigrquery.r-dbi.org/), [googledrive](https://googledrive.tidyverse.org/), and [googlesheets4](https://googlesheets4.tidyverse.org/).[^2]
+
+[^2]: gmailr is not included here, because the Gmail scopes are so sensitive that anyone wishing to work with Gmail programmatically really must set up their own OAuth client.
+
+As of gargle v1.0.0, released in March 2021, we have rolled the "Tidyverse API Packages" client, i.e. we are using a new one.
+Although we have not yet disabled the previous client, users should be prepared for it to stop working at any time.
```

Comment:
I wonder if we should set a date for this 🤔

Reply:
If we do so, it probably needs to be its own blog post or VERY PROMINENT in this one.

Reply:
One dilemma is that I suspect many of the people who need to know ... don't even know they need to know.

That is, disabling to the old client is going to cause some pain and I don't think there's any way to prevent most of it, regardless of the announcement strategy.

But yeah, I agree that a separate blog post makes sense, in addition to hammering on this here and in future posts for googledrive/googlesheets4/bigrquery.

Reply:
I've mentioned December 31, 2021 as the date now and alluded to a future blog post.